fili-core/src/main/java/com/yahoo/bard/webservice/druid/util/ThetaSketchFieldConverter.java

```diff
@@ -30,21 +31,45 @@ public  ThetaSketchAggregation asInnerSketch(SketchAggregation candidate) {
         return new ThetaSketchAggregation(candidate.getName(), candidate.getFieldName(), candidate.getSize());
     }
 
-   @Override
-    public ThetaSketchEstimatePostAggregation asSketchEstimate(SketchAggregation candidate) {
+    /**
+     * Get the candidate Aggregation as a SketchEstimatePostAggregation.
+     *
+     * @param candidate  Aggregation to "convert"
+     *
+     * @return The Aggregation as a SketchEstimatePostAggregation
+     */
+    public ThetaSketchEstimatePostAggregation asSketchEstimate(Aggregation candidate) {
         String name = candidate.getName() + "_estimate";
         PostAggregation field = new FieldAccessorPostAggregation(candidate);
         return new ThetaSketchEstimatePostAggregation(name, field);
     }
 
+    /**
+     * Get the PostAggregation as a SketchEstimatePostAggregation.
+     *
+     * @param postAggregation  PostAggregation to "convert"
+     *
+     * @return The PostAggregation as a SketchEstimatePostAggregation
+     */
+    public ThetaSketchEstimatePostAggregation asSketchEstimate(PostAggregation postAggregation) {
+        String name = postAggregation.getName() + "_estimate";
+        return new ThetaSketchEstimatePostAggregation(name, postAggregation);
+    }
```

Comment:
I'm little concerned about these three overloadings:

`asSketchEstimate(Aggregation candidate)`
`asSketchEstimate(PostAggregation postAggregation)`
`asSketchEstimate(MetricField field)`

Given that both `Aggregation` and `PostAggregation` implements `MetricField`, consider this case:

```java
List<MetricField> fields = [new xyzAggregation(...), new xyzPostAggregation(...), new xyzMetricField(...)]
for (MetricField field : fields) {
    thetaSketchFieldConverter.asSketchEstimate(field)
}
```

I suspect the case above will end up calling `asSketchEstimate(MetricField field)` for all 3 iterations although `xyzPostAggregation` should really call `asSketchEstimate(PostAggregation postAggregation)` and `xyzAggregation` should really call `asSketchEstimate(Aggregation candidate)`

The choice of which overloading to invoke is made at compile time. For all three iterations of the loop, the compile-time type of the parameter is the same: `MetricField`. The runtime type is different in each iteration, but this does not affect the choice of overloading. Because the compile-time type of the parameter is `MetricField`, the only applicable overloading is the third one, `asSketchEstimate(MetricField field)`, and this overloading is invoked in each iteration of the loop.

__One solution is not exporting two overloadings with the same number of parameters.__